pgx/src/spi.rs

```diff
@@ -487,6 +488,48 @@ impl SpiTupleTable {
             }
         }
     }
+
+    /// Returns the number of columns
+    pub fn columns(&self) -> Option<i32> {
+        match self.tupdesc {
+            Some(tupdesc) => unsafe { Some((*tupdesc).natts) },
+            None => None,
+        }
+    }
+
+    /// Returns column type OID
+    pub fn column_type_oid(&self, ordinal: i32) -> Option<PgOid> {
+        match self.tupdesc {
+            Some(tupdesc) => unsafe {
+                let nattrs = (*tupdesc).natts;
+                if ordinal < 1 || ordinal > nattrs {
+                    None
+                } else {
+                    let oid = pg_sys::SPI_gettypeid(tupdesc, ordinal);
+                    Some(PgOid::from(oid))
+                }
+            },
+            None => None,
+        }
+    }
+
+    /// Returns column name
+    pub fn column_name(&self, ordinal: i32) -> Option<String> {
+        match self.tupdesc {
+            Some(tupdesc) => unsafe {
+                let nattrs = (*tupdesc).natts;
+                if ordinal < 1 || ordinal > nattrs {
+                    None
+                } else {
+                    let name = pg_sys::SPI_fname(tupdesc, ordinal);
```

Comment:
Would it be less surprising to adjust this to 0-indexed?

Reply:
I am following that's already in Spi and Postgres itself.

Reply:
It's probably worth mentioning in the documentation comment.

Reply:
@thomcc I've updated the documentation to reflect this.